site/content/contribute/desktop/debugging.md

```diff
@@ -6,16 +6,48 @@ date: 2019-01-22T00:00:00-05:00
 weight: 3
 ---
 
-The electron app itself can be inspected using the developer tools, available from the View menu:
+## Debugging in the Main Process
 
-![View toggle developer tools](/contribute/desktop/view-toggle-developer-tools.png)
+The simplest way to debug the main process is to simply insert logging statements wherever needed and having the application output logs of whatever is necessary.
 
-However, this view does not expose the contents of the webviews for inspection. To open the developer tools for a webview, first access the console for the electron shell as above, and then run
+If you'd like to make use of better debugging tools, you can make use of the Chrome Dev Tools or the debugger in VSCode by following the steps here: https://www.electronjs.org/docs/latest/tutorial/debugging-main-process
 
-```js
-document.getElementById("mattermostView0").openDevTools();
-```
+## Debugging in the Renderer Process
 
-Use `mattermostView0` to inspect the first opened tab, `mattermostView1` to inspect the second opened tab, and so on. A new window will appear with the developer tools for that webview:
+The renderer processes are controller by Chrome instances, so each of them will have their own Developer Tools instance.
+
+You can access these instances by going to the View menu (under the 3-dot menu on Windows/Linux, and in the top bar on macOS) and selecting:
+- **Developer Tools for Application Wrapper** for anything involving the top bar.
+- **Developer Tools for Current Tab** for anything involving the Mattermost view or the preload script
+    - *Note:* for this one you need to make sure you're currently on the tab you want to load the Developer Tools for. You can have instances of the tools open for tabs you aren't currently viewing, but to open them in the first place requires it to be opened.
```

Comment:
```suggestion
    - **Note:** For this one, make sure you're currently on the tab where you want to load the Developer Tools. You can have instances open for tabs you aren't currently viewing, but to open them in the first place requires it to be opened.
```

Reply:
I'm struggling a bit with parsing this sentence: "You can have instances open for tabs you aren't currently viewing, but to open them in the first place requires it to be opened." The first part is clear; the second part is less clear.